csharp/ql/src/Security Features/CWE-502/UnsafeDeserializationUntrustedInput.ql

```diff
@@ -13,8 +13,54 @@
 import csharp
 import semmle.code.csharp.security.dataflow.UnsafeDeserialization::UnsafeDeserialization
 import DataFlow::PathGraph
+import semmle.code.csharp.security.dataflow.flowsources.Remote
+import semmle.code.csharp.security.dataflow.flowsources.Local
 
-from TaintTrackingConfig config, DataFlow::PathNode source, DataFlow::PathNode sink
-where config.hasFlowPath(source, sink)
-select sink.getNode(), source, sink, "$@ flows to unsafe deserializer.", source.getNode(),
+class RemoteSource extends Source {
+  RemoteSource() { this instanceof RemoteFlowSource }
+}
+
+class LocalSource extends Source {
+  LocalSource() { this instanceof LocalFlowSource }
+}
+
+from
+  TaintToObjectMethodTrackingConfig taintTracking, DataFlow::PathNode userInput,
+  DataFlow::PathNode deserializeCall
```

Comment:
`UnsafeDeserialization.ql` introduces `Call deserializeCall`, which is really a call. While here `DataFlow::PathNode deserializeCall` is not a call, but rather an argument of a call, isn't it? This is quite misleading. I would rename `deserializeCall` to `deserializeCallArg`.